cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -216,29 +233,174 @@ def _update_postgres_password(self):
+                    etcd_password=cluster_config['etcd']['patroni_password'],
+                    etcdctl=' '.join(etcdctl_base_command)
+                ),
+                shell=True,
+            )
+            common.run(
+                etcdctl_base_command +
+                ['user', 'grant', 'patroni', '--roles', 'guest']
+            )
+            common.run(etcdctl_base_command + ['auth', 'enable'])
+
+        logger.info('Creating postgres bin links for patroni')
+        for pg_bin in PG_BINS:
+            common.sudo(['ln', '-s', os.path.join(PG_BIN_DIR, pg_bin),
+                         '/usr/sbin'])
+
+        logger.info('Starting patroni')
+        files.deploy(
+            os.path.join(CONFIG_PATH, 'patroni.service'),
+            '/usr/lib/systemd/system/patroni.service',
+            render=False,
+        )
+        systemd.enable('patroni', append_prefix=False)
+        systemd.start('patroni', append_prefix=False)
 
     def install(self):
         logger.notice('Installing PostgreSQL Server...')
-        self._install()
+        sources = config[POSTGRESQL_SERVER][SOURCES]
+
+        logger.debug('Installing PostgreSQL Server dependencies...')
+        yum_install(sources['libxslt_rpm_url'])
+
+        logger.debug('Installing PostgreSQL Server...')
+        yum_install(sources['ps_libs_rpm_url'])
+        yum_install(sources['ps_rpm_url'])
+        yum_install(sources['ps_contrib_rpm_url'])
+        yum_install(sources['ps_server_rpm_url'])
+        yum_install(sources['ps_devel_rpm_url'])
+        # As we don't support installing community as anything other than AIO,
+        # not having manager service installed means that this must be premium
+        if MANAGER_SERVICE not in config[SERVICES_TO_INSTALL]:
+            yum_install(sources['etcd_rpm_url'])
+            yum_install(sources['patroni_rpm_url'])
         logger.notice('PostgreSQL Server successfully installed')
 
     def configure(self):
         logger.notice('Configuring PostgreSQL Server...')
-        self._configure()
+        files.copy_notice(POSTGRESQL_SERVER)
+
+        if config[POSTGRESQL_SERVER]['cluster']['nodes']:
+            self._configure_cluster()
+        else:
+            self._init_postgresql_server()
+            enable_remote_connections = \
+                config[POSTGRESQL_SERVER][ENABLE_REMOTE_CONNECTIONS]
+            self._update_configuration(enable_remote_connections)
+            if config[POSTGRESQL_SERVER][POSTGRES_PASSWORD]:
+                self._update_postgres_password()
+
+            systemd.restart(SYSTEMD_SERVICE_NAME, append_prefix=False)
+            systemd.verify_alive(SYSTEMD_SERVICE_NAME, append_prefix=False)
         logger.notice('PostgreSQL Server successfully configured')
 
     def remove(self):
+        if config[POSTGRESQL_SERVER]['cluster']['nodes']:
+            logger.notice('Removing cluster components')
+            yum_remove('etcd')
```

Comment:
those are installed if it's a db-only node, not if `config[postgres][cluster][nodes]`, right?
the install check and the remove check should be the same

Reply:
Fixed.